lib/qam.pm

```diff
@@ -80,8 +82,28 @@ sub add_repo_if_not_present {
     zypper_call("--no-gpg-checks ar -f $gpg -n '$name' $url '$name'") unless grep { $_->{uri} eq $url } @$system_repos;
 }
 
+# https://progress.opensuse.org/issues/131498
+sub add_specific_extra_customer_repositories {
+    $specific_extra_repo = 1;
+    my $arch = get_var('ARCH');
+    my %repo_list = (
+        x86_64 => [
+            {cond => '=15-SP4', name => '15-SP4-TERADATA-Updates', uri => 'http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP4-TERADATA/x86_64/update/', customer => 'TERADATA'},
+        ],
+        s390x => [],
+        ppc64le => [],
+        aarch64 => [],
+    );
+
+    return unless $repo_list{$arch};
+    for my $repo (@{$repo_list{$arch}}) {
+        add_repo_if_not_present($repo->{uri}, $repo->{name}) if is_sle($repo->{cond}) && $repo->{name} =~ /$repo->{customer}/;
```

Comment:
`$repo->{name} =~ /$repo->{customer}/` will be either always true or always false, this doesn't make sense.